main.py

```diff
@@ -15,41 +12,121 @@
 
 def get_time(slot):
-    # Obtains time for each slot from 'data/slots.1' file
-
+    """Obtains time for each slot from 'data/slots.1' file"""
     with open(os.path.join(path, "data/slots.1")) as f:
         for line in f:
             if line.startswith(slot):
                 return line.split()[1:]
 
+def parse_html(dep, session):
+    """Parses HTML to get professor information and their timetable"""
+    headers = {
+        'Accept': 'application/json, text/javascript, */*; q=0.01',
+        'Accept-Encoding': 'gzip, deflate, br, zstd',
+        'Accept-Language': 'en-GB,en;q=0.9',
+        'Connection': 'keep-alive',
+        'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+        'Cookie': 'ci_session=gk7bi3u0r2ej0h3qnihk4gocf0vp9e8e',
+        'Host': 'www.iitkgp.ac.in',
+        'Origin': 'https://www.iitkgp.ac.in',
+        'Referer': 'https://www.iitkgp.ac.in/faclistbydepartment',
+        'Sec-Fetch-Dest': 'empty',
+        'Sec-Fetch-Mode': 'cors',
+        'Sec-Fetch-Site': 'same-origin',
+        'Sec-GPC': '1',
+        'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36',
+        'X-Requested-With': 'XMLHttpRequest',
+        'sec-ch-ua': '"Not/A)Brand";v="8", "Chromium";v="126", "Brave";v="126"',
+        'sec-ch-ua-mobile': '?0',
+        'sec-ch-ua-platform': '"Linux"'
+    }
+
+    start = 0
+    length = 10
+    more_pages = True
+
+    while more_pages:
+        payload = {
+            'draw': 1,
+            'columns[0][data]': 'empname',
+            'columns[0][name]': '',
+            'columns[0][searchable]': 'true',
+            'columns[0][orderable]': 'true',
+            'columns[0][search][value]': '',
+            'columns[0][search][regex]': 'false',
+            'columns[1][data]': 'department',
+            'columns[1][name]': '',
+            'columns[1][searchable]': 'true',
+            'columns[1][orderable]': 'true',
+            'columns[1][search][value]': '',
+            'columns[1][search][regex]': 'false',
+            'columns[2][data]': 'designation',
+            'columns[2][name]': '',
+            'columns[2][searchable]': 'true',
+            'columns[2][orderable]': 'true',
+            'columns[2][search][value]': '',
+            'columns[2][search][regex]': 'false',
+            'order[0][column]': '0',
+            'order[0][dir]': 'asc',
+            'start': str(start),
+            'length': str(length),
+            'search[value]': '',
+            'search[regex]': 'false',
+            'lang': 'en'
+        }
+
+        dept_resp = session.post(DEPT_FETCH_URL, headers=headers, data=payload)
+        dept_raw_data = json.loads(dept_resp.content).get("data", [])
+
+        if not dept_raw_data:
+            more_pages = False
+        else:
+            for prof in dept_raw_data:
+                name = re.findall(r">(.+?)<", prof["faculty"])[0].replace("  ", "")
```

Comment:
Can you add some comments explaining the code?

Reply:
Okay, I add comments and update the code.